src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/MessageProcessor.java

```diff
@@ -7,26 +7,26 @@
 import org.slf4j.LoggerFactory;
 import org.springframework.scheduling.annotation.Scheduled;
 import org.springframework.stereotype.Service;
-import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.ReceiverProvider;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.servicebus.MessageReceiverFactory;
 
 @Service
 public class MessageProcessor {
 
     private static final Logger logger = LoggerFactory.getLogger(MessageProcessor.class);
 
-    private final ReceiverProvider receiverProvider;
+    private final MessageReceiverFactory receiverProvider;
```

Comment:
field name no longer matches type name. but... was this class name necessary? (and method name rename)

Reply:
Its now injecting via an interface ... so it can be replaced with a bean in the integration test.
Ideally we would not need to do this but since we are using shared infrastructure for testing we are forced to mock this out otherwise it would be too fragile.